example/gardener-local/gardenlet/values.yaml

```diff
@@ -59,6 +59,19 @@ config:
       - "production"
       - "development"
       - "evaluation"
+  etcdConfig:
+    etcdController:
+      workers: 3
+    custodianController:
+      workers: 3
+    backupCompactionController:
+      workers: 3
+      enableBackupCompaction: true
+      eventsThreshold: 10000
+      activeDeadlineDuration: "3h"
+    backupLeaderElection:
+      reelectionPeriod: 5s
+      etcdConnectionTimeout: 5s
```

Comment:
I think this file is used for e2e tests and local setup. Can you elaborate why the default values don't work and we have to specify the above values?

Reply:
`etcdController.workers` has default value 50, `custodianController.workers` has default value 10, `backupCompactionController. workers` has already default value 3, `backupCompactionController. enableBackupCompaction` has default value false, `backupCompactionController.eventsThreshold` has default value 1000000, I don't see any defaulting in g/g for `backupCompactionController.activeDeadlineDuration` but its doc string says that it default to 3h. For `backupLeaderElection` fields I don't see any defaulting in g/g.

https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/pkg/gardenlet/apis/config/v1alpha1/defaults.go#L474-L500

From the context I have, I guess there is need only to set `backupCompactionController.enableBackupCompaction` and `backupCompactionController.eventsThreshold`. I guess it does not harm to have high values for the `workers` values. 

I am writing all of this as the pattern that seems to be followed in this file is to specify only the values which are different from the defaults.

For example, only 2 values are specified for gardenlet controllers https://github.com/gardener/gardener/blob/f81703bda20f5be17356b1d2f4cd9c740b3a9ea2/example/gardener-local/gardenlet/values.yaml#L32-L36 while while there are actually many controllers with different kind of configs - https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/example/20-componentconfig-gardenlet.yaml#L17-L89.

---

I would also ask why we miss defaulting for the gardenlet component config for fields like `backupCompactionController.activeDeadlineDuration` and `backupLeaderElection`? Is there a specific reason?

Reply:
I raised this PR as it's helpful for a new user to easily set druid config. Yes, the values can have same values as the default value but if somebody wants to test something, they can easily change the values in this file and test the druid. I already got such requests from people to show how to tweak the `eventhreshold` value for compaction in local setup. I thought, this PR would be good to have

Reply:
I'd agree it's fair not to set values that aren't needed here. We don't need to include `etcdController` and `custodianController` in the config, but we can stick with `backupCompactionController`

Reply:
> I would also ask why we miss defaulting for the gardenlet component config for fields like backupCompactionController.activeDeadlineDuration and backupLeaderElection? Is there a specific reason?

From [here](https://github.com/gardener/gardener/blob/master/pkg/component/etcd/bootstrap.go#L400-L401) I can see that we don't want to set the `--active-deadline-duration` flag for etcd druid if `backupCompactionController.activeDeadlineDuration` is not set. So ig it makes sense not to have a default value

I didn't check the code for `backupLeaderElection`, however I would assume it's the same reason